SConstruct

```diff
@@ -586,25 +609,138 @@ def list_sources(base, suffixes):
+             'src/sqlite',]
+    )
+    append_sources(result,
+        *list_sources('src/ripple/basics', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/core', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/crypto', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/json', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/legacy', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/net', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/overlay', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/peerfinder', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/protocol', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/shamap', '.cpp'))
+    append_sources(result,
+        *list_sources('src/ripple/nodestore', '.cpp'),
+        CPPPATH=[
+            'src/rocksdb2/include',
+            'src/snappy/snappy',
+            'src/snappy/config',
+        ])
+
+    result += get_soci_sources('classic')
+    return result
+
+def get_unity_sources():
+    result = []
+    append_sources(result,
+        'src/ripple/unity/app.cpp',
+        'src/ripple/unity/app1.cpp',
+        'src/ripple/unity/app2.cpp',
+        'src/ripple/unity/app3.cpp',
+        'src/ripple/unity/app4.cpp',
+        'src/ripple/unity/app5.cpp',
+        'src/ripple/unity/app6.cpp',
+        'src/ripple/unity/app7.cpp',
+        'src/ripple/unity/app8.cpp',
+        'src/ripple/unity/app9.cpp',
+        'src/ripple/unity/core.cpp',
+        'src/ripple/unity/basics.cpp',
+        'src/ripple/unity/crypto.cpp',
+        'src/ripple/unity/net.cpp',
+        'src/ripple/unity/overlay.cpp',
+        'src/ripple/unity/peerfinder.cpp',
+        'src/ripple/unity/json.cpp',
+        'src/ripple/unity/protocol.cpp',
+        'src/ripple/unity/shamap.cpp',
+        'src/ripple/unity/legacy.cpp',
+    )
+
+    result += get_soci_sources('unity')
+
+    append_sources(result,
+        'src/ripple/unity/nodestore.cpp',
+        CPPPATH=[
+            'src/rocksdb2/include',
+            'src/snappy/snappy',
+            'src/snappy/config',
+        ])
+
+    return result
 
 # Declare the targets
 aliases = collections.defaultdict(list)
 msvc_configs = []
 
+def should_prepare_target (clTarget,
+                           style, toolchain, variant):
+    if not clTarget:
+        # default target
+        return (style == default_tu_style and
+            toolchain == default_toolchain and variant == default_variant)
+    if 'vcxproj' in clTarget:
+        return True
+    s = clTarget.split('.')
+
+    if 'unity' == 'style' and 'nounity' in s:
+        return False
+
+    ls = len(s)
+    if (1 == ls):
+        return 'all' in clTarget or variant in clTarget or toolchain in clTarget
+    if (2 == ls):
+        return s[0] == toolchain and s[1] == variant
+
+    return True # should never happen
```

Comment:
If it should never happen, why are you writing code for it?

    assert len(s) > 2, 'Target had too many commas: "%s"' % clTarget
    if len(s) == 1:
        return 'all' in clTarget or variant in clTarget or toolchain in clTarget
    return s[0] == toolchain and s[1] == variant